src/shared/logger/logger.ts

```diff
@@ -18,9 +18,38 @@ export interface Logger {
     warn(error: Error): void
     error(message: string, ...meta: any[]): void
     error(error: Error): void
+    /** Returns true if the given log level is being logged.  */
+    logLevelEnabled(logLevel: LogLevel): boolean
 }
 
-export type LogLevel = keyof Logger
+/**
+ * Log levels ordered for comparison.
+ *
+ * See https://github.com/winstonjs/winston#logging-levels :
+ * > RFC5424: severity of levels is numerically ascending from most important
+ * > to least important.
+ */
+const logLevels = new Map<LogLevel, number>([
+    ['error', 1],
+    ['warn', 2],
+    ['info', 3],
+    ['verbose', 4],
+    ['debug', 5],
+])
+
+export type LogLevel = 'error' | 'warn' | 'info' | 'verbose' | 'debug'
```

Comment:
The old `export type LogLevel = keyof Logger` approach means the `Logger` interface can never have any other methods on it. The cost of that already can be seen, because for example it is missing things like `setLogLevel()`, which leads to awkward code that requires casting (defeats the purpose of an interface).